src/test/py/bazel/bzlmod/bazel_module_test.py

```diff
@@ -364,6 +364,38 @@ def testRepositoryRuleErrorInModuleExtensionFailsTheBuild(self):
             in line for line in stderr
         ]))
 
+  def testHttpfileModuleExtensionSuccess(self):
+      self.ScratchFile('MODULE.bazel', [
+          'file_ext = use_extension("//tools/build_defs/repo:http.bzl", "http_file_ext")',
```

Comment:
If the extension were just called `http`, this would make it easy to add an `archive` tag class later.

Reply:
Good point